src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpCodeStyleOptions.cs

```diff
@@ -57,6 +57,13 @@ private static Option2<T> CreateOption<T>(OptionGroup group, string name, T defa
                 EditorConfigStorageLocation.ForBoolCodeStyleOption("csharp_style_prefer_switch_expression"),
                 new RoamingProfileStorageLocation($"TextEditor.CSharp.Specific.{nameof(PreferSwitchExpression)}")});
 
+        public static readonly Option2<CodeStyleOption2<bool>> PreferPatternMatching = CreateOption(
+            CSharpCodeStyleOptionGroups.PatternMatching, nameof(PreferPatternMatching),
+            defaultValue: s_trueWithSuggestionEnforcement,
```

Comment:
havingthis be on with suggestion is likely a non-starter IMO.  This is such idiomatic code, we would end up suggesting tons of times in every codebase.  And it's not clear to me that the new form is strictly superior.  So i think this needs to be hidden by default, but users can opt into it.